packages/roosterjs-editor-plugins/lib/plugins/HyperLink/HyperLink.ts

```diff
@@ -67,6 +76,43 @@ export default class HyperLink implements EditorPlugin {
      * @param event PluginEvent object
      */
     public onPluginEvent(event: PluginEvent): void {
+        if (
+            event.eventType == PluginEventType.MouseUp ||
+            (event.eventType == PluginEventType.KeyUp &&
+                event.rawEvent.which >= Keys.PAGEUP &&
+                event.rawEvent.which <= Keys.DOWN) ||
+            event.eventType == PluginEventType.ContentChanged
+        ) {
+            const anchor = this.editor.getElementAtCursor(
+                'A[href]',
+                null /*startFrom*/,
+                event
+            ) as HTMLAnchorElement;
+
+            // If cursor has moved out of previously tracked link
+            // update link href if applicable and then reset tracked state
+            if (this.trackedLink && anchor !== this.trackedLink) {
+                if (!this.doesLinkDisplayMatchHref(this.trackedLink)) {
+                    this.updateLinkHref(event);
+                }
+
+                this.trackedLink = null;
+                this.originalHref = '';
+            }
+
+            // If the link's href value was edited, stop tracking the link.
+            if (this.trackedLink && this.tryGetHref(this.trackedLink) !== this.originalHref) {
+                this.trackedLink = null;
+                this.originalHref = '';
+            }
```

Comment:
These two if blocks (L94-L101 and L104-L107) seems to be able to merge.

Reply:
I did some refactoring to combine logic with the blur event as well. I could combine the logic further so that the link tracking and updating is all done in one function and both the onBlur event and the onPluginEvents could call into it, but I feel that it would make the logic more complicated since it would mean having many optional fields for that function.

Let me know what you think.

Reply:
Current code seems good to me.